vm/src/obj/objframe.rs

```diff
@@ -30,13 +30,21 @@ impl FrameRef {
     }
 
     #[pyproperty]
-    fn f_globals(self, _vm: &VirtualMachine) -> PyDictRef {
-        self.scope.globals.clone()
+    fn f_globals(self, vm: &VirtualMachine) -> PyResult<PyDictRef> {
+        if self.code.incognito {
```

Comment:
Is this nessessary to check this option every time in runtime?

Reply:
Yes; the incognito option is on the `CodeObject` and not the `Scope`, so we have to do the check at the point where the two meet, which is the frame/function object.